### PR TITLE
Move JS dependencies into npm and support hot module replacement

### PR DESCRIPTION
**Pull request by KorvinSzanto**
If you're not familiar with HMR, it allows you to save a resource file like a js or css file and see those changes show up in the browser nearly immediately:
hmr-example

This method of development for JS and CSS is a lot quicker than building manually or using the watch script, and is supported using the `ui.mix(...)` function added to the ui service.

By the way this PR has less in it than it looks, once #67 merges this PR will have less modifications.

### CODE REVIEW

**Comment by KorvinSzanto** [comment]
I'm confused why the tests would fail on this? I don't see the failure.

File: ui/ui.go [hunk 1]
```diff
@@ -258,7 +262,8 @@ func (h *uiHandler) Certificate() http.HandlerFunc {
 
 		head := headTemplate{
 			fmt.Sprintf("Certificate - %s", cert.CommonName),
-			"site.css",
+			h.mix("/css/site.css"),
+			h.mix("/js/site.js")
```
Comment (d1str0) [inline ui/ui.go]: Compilation error here, need trailing comma.

**Review by d1str0 — CHANGES_REQUESTED** [review]

**Comment by d1str0** [comment]
BTW, if you run 'go test ./...' in the root dir for the project, you should see any compilation errors.

File: ui/ui.go (hunk 1 above)
Comment (KorvinSzanto) [reply ui/ui.go]: What makes it require the trailing comma? Is it the multiline?

**Comment by KorvinSzanto** [comment]
I didn't see the error when I ran the circleci test locally or when I reviewed the circle ci log, can you point it out to me?

**Review by domshyra — DISMISSED** [review]

**Review by d1str0 — APPROVED** [review]
LGTM

**Pull request merged**